garage/tf/policies/discrete_qf_derived_policy.py

```diff
@@ -0,0 +1,75 @@
+"""
+QfDerived policy.
+
+This policy chooses the action that yields to the largest q-value.
+"""
+import numpy as np
+import tensorflow as tf
+
+from garage.core import Serializable
+from garage.misc.overrides import overrides
+from garage.spaces import Discrete
+from garage.tf.policies import Policy
+
+
+class DiscreteQfDerivedPolicy(Policy, Serializable):
+    """
+    DiscreteQfDerived policy.
+
+    Args:
+        env_spec: Environment specification.
+        qf: The q-function used.
+    """
+
+    def __init__(self, env_spec, qf):
+        Serializable.quick_init(self, locals())
+        super().__init__(env_spec)
+
+        assert isinstance(env_spec.action_space, Discrete)
+        self._env_spec = env_spec
+        self._qf = qf
+
+    @property
+    def vectorized(self):
+        """Vectorized or not."""
+        return True
+
+    @overrides
+    def get_action(self, observation):
+        """
+        Get action from this policy for the input observation.
+
+        Args:
+            observation: Observation from environment.
+            sess: tf.Session provided.
+
+        Returns:
+            opt_action: Optimal action from this policy.
+
+        """
+        sess = tf.get_default_session()
```

Comment:
Is this gonna be a problem? Have we settled down how do we design this?

Reply:
AFAICT this should work fine now. I don't think we settled that. Can you propose some designs?

Reply:
Yeah, I agree it works properly now. Since we introduced Model class and made the assumption that a session is created when we instantiate a primitive, we can later replace this with make_callable.